packages/core/echo/echo-schema/src/database.ts

```diff
@@ -110,6 +110,11 @@ export class EchoDatabase {
       obj[base]._beforeBind();
 
       const snapshot = obj[base]._createSnapshot();
+
+      log('add to set', { id: obj[base]._id, instance: getDebugName(obj) });
+      invariant(!this._objects.has(obj[base]._id));
+      this._objects.set(obj[base]._id, obj);
```

Comment:
This was causing issues if it was called before `obj[base]._beforeBind()` (line 110) call, so this fix should not cause issues